aspnet/fundamentals/routing.rst

```diff
@@ -2,81 +2,201 @@ Routing
+
+See route-template-reference_ for a thorough description of route template features and syntax.
+
+This example includes a route constraint:
+
+.. code-block:: c#
+
+    routes.MapRoute(
+        name: "default",
+        template: "{controller=Home}/{action=Index}/{id:int}");
+
+This template will match a URL path like ``/Products/Details/17``, but not ``/Products/Details/Apples``. The route parameter definition ``{id:int}`` defines *route constraint* for the ``id`` route parameter. Route constraints implement ``IRouteConstraint`` and inspect route values to verify them. In this example the route value ``id`` must be convertable to an integer. See route-constraint-reference_ for a more detailed explaination of route constraints that are provided by the framework.
+
+Additional overloads of ``MapRoute`` accept values for ``constraints``, ``dataTokens``, and ``defaults``. These additional parameters of ``MapRoute`` are defined as type ``object``. The typical usage of these parameters is to pass an anonymously typed object, where the property names of the anonymous type match route parameter names.
+
+The following two examples create equivalent routes:
+
+.. code-block:: c#
+
+    routes.MapRoute(
+        name: "default_route",
+        template: "{controller}/{action}/{id?}", 
+        defaults: new { controller = "Home", action = "Index" });
+
+    routes.MapRoute(
+        name: "default_route",
+        template: "{controller=Home}/{action=Index}/{id?}");
+
+.. tip:: The inline syntax for defining constraints and defaults can be more convenient for simple routes. However, there are features such as data tokens which are not supported by inline syntax.
+
+This example demonstrates a few more features:
+
+.. code-block:: c#
+
+    routes.MapRoute(
+        name: "blog",
+        template: "/Blog/{*article}",
+        defaults: new { controller = "Blog", action = "ReadArticle" });
+
+This template will match a URL path like ``/Blog/All-About-Routing/Introduction`` and will extract the values ``{ controller = Blog, action = ReadArticle, article = All-About-Routing/Introduction }``. The default route values for ``controller`` and ``action`` are produced by the route even though there are no corresponding route parameters in the template. Default values can be specified in the route template The ``article`` route parameter is defined as a *catch-all* by the appearance of an asterix ``*`` before the route parameter name. Catch-all route parameters capture the remainder of the URL path, and can also match the empty string.
+
+This example adds route constraints and data tokens:
+
+.. code-block:: c#
+
+    routes.MapGet(
+        name: "us_english_products",
+        template: "/en-US/Products/{id}",
+        defaults: new { controller = "Products", action = "Details" },
+        constraints: new { id = new IntConstraint() },
```

Comment:
@rynowak  can I change that to IntRouteConstraint()
https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Routing/Constraints/IntRouteConstraint/index.html